sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySourceDef.java

```diff
@@ -42,15 +40,15 @@ interface BigQuerySourceDef extends Serializable {
   <T> BigQuerySourceBase<T> toSource(
       String stepUuid,
       Coder<T> coder,
-      SerializableFunction<TableSchema, AvroSource.DatumReaderFactory<T>> readerFactory,
+      BigQueryReaderFactory<T> readerFactory,
       boolean useAvroLogicalTypes);
 
   /**
-   * Extract the Beam {@link Schema} corresponding to this source.
+   * Extract the {@link TableSchema} corresponding to this source.
    *
    * @param bqOptions BigQueryOptions
-   * @return Beam schema of the source
+   * @return table schema of the source
    * @throws BigQuerySchemaRetrievalException if schema retrieval fails
    */
-  Schema getBeamSchema(BigQueryOptions bqOptions);
+  TableSchema getTableSchema(BigQueryOptions bqOptions);
```

Comment:
As this is in BQ realm, it makes more sense to return unaltered `TableSchema`

Reply:
There was proposal to make Beam Schema uniform in future Beam versions (3.x), unfortunately BigQueryIO is the biggest IO that do not follow this (and has its own TableSchema that does not implements Serializable). I would suggest keep  "getBeamSchema". We can have "getTableSchema" as addition.